From bdec38b799fc9499ffe51f5cb2e42dfec9c414e7 Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 30 Mar 2017 09:13:43 +0800
Subject: [PATCH 065/347] ARM: dts: keystone-k2g: Add device state controller
 node

This patch comes from:
  git://git.ti.com/processor-sdk/processor-sdk-linux.git

commit 7a8f113f3e2b2803148a2033c21fc98ca706b2a0 ti-sdk

Add the device state controller node as a syscon node to the
Keystone 2 Galileo SoC. This module provides similar device
control functionality as that on the existing K2 SoCs.

One example usage would be the boot address programming of the
DSP processor sub-system.

Signed-off-by: Andrew F. Davis <afd@ti.com>
[s-anna@ti.com: relocate node and revise commit description, 4.4 port]
Signed-off-by: Suman Anna <s-anna@ti.com>
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 arch/arm/boot/dts/keystone-k2g.dtsi |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/keystone-k2g.dtsi b/arch/arm/boot/dts/keystone-k2g.dtsi
index 3bac719..021748f 100644
--- a/arch/arm/boot/dts/keystone-k2g.dtsi
+++ b/arch/arm/boot/dts/keystone-k2g.dtsi
@@ -80,6 +80,11 @@
 		ranges = <0x0 0x0 0x0 0xc0000000>;
 		dma-ranges = <0x80000000 0x8 0x00000000 0x80000000>;
 
+		devctrl: device-state-control@02620000 {
+			compatible = "ti,keystone-devctrl", "syscon";
+			reg = <0x02620000 0x1000>;
+		};
+
 		k2g_pinctrl: pinmux@02621000 {
 			compatible = "pinctrl-single";
 			reg = <0x02621000 0x410>;
-- 
1.7.5.4

